### ISSUE TITLE
AzureBlobStorage-hosted AdaptiveCard image-rendering broken on Teams iOS

### ISSUE THREAD

**Issue by aimskl** [MicrosoftDocs/msteams-docs#3124]
I wasn't sure where to post this issue/bug report, so I am posting it here in the msteams-docs area.

I also posted about this a few days ago here: https://microsoftteams.uservoice.com/forums/908686-bug-reports/suggestions/43151277-image-in-adaptive-card-on-teams-ios-no-longer-rend

**_Background_**
We use a Logic App to send an hourly Adaptive Card to a Teams channel webhook. The Adaptive Card contains some text and an image (hosted in azure blob storage using the Blob (anonymous read access for blobs only) public access level). This public azure blob storage solution for storing the images for our AdaptiveCard has worked smoothly for at least three years.

As of 2021-04-12 11:40 UTC, our AdaptiveCard image stopped getting rendered in Teams on iOS. Instead we see the broken image placeholder. However, the image still works fine when viewing the adaptive cards using Teams in browser or desktop.

We have not changed our AdaptiveCard json payload structure/config in over a year. 

**_Further self-troubleshooting_**
For the purposes of demonstrating the problem here, we created a new storage account with the exact same settings as our production environment. Storage account is GenV1.

The url to the testing image that is included in the AdaptiveCard is as follows: https://4rpm16vw6eyxvm4z05tvv4yr.blob.core.windows.net/99fpycl6ri99x2f/1000.png
{We will delete this test storage account once this problem has been acknowledged/replicated}

As mentioned, this image renders perfectly in the AdaptiveCard when viewed via Teams Desktop or Teams Web Browser. However, since the above mentioned timestamp, it has stopped working on Teams iOS (not sure about other mobile platforms).

We first thought maybe Teams had changed the restrictions on image size (dimensions or file size), but after checking the docs we confirmed that the production image is consistently these dimensions and rough filesize, so, well within the restrictions:
![image](https://user-images.githubusercontent.com/71928381/114992779-b3629a00-9ecd-11eb-81d9-1ead9e662dc1.png)
Conclusion: the problem is not image size.

We then noticed in the blob storage account a spike in 'AuthenticationError' and 'ClientOtherError' errors starting from the above mentioned timestamp, and also corresponding with our testing as described here. The errors do not appear when access to the image was made from Teams Desktop or Browser; rather, only when access was attempted from Teams iOS.
![image](https://user-images.githubusercontent.com/71928381/114992264-261f4580-9ecd-11eb-95a3-3753b2d567c1.png)

We tried turning off Secure Access (https) and reducing minimum TLS to 1.0 for this blob account, but that made no difference.

Finally, we tried hosting the same test image shown at the url above from a different place. So we used the link from where we originally got the image, which is: https://via.placeholder.com/1000
Lo and behold, this caused Teams iOS to render the image with no problem. (likewise for Desktop/WebBrowser Teams). Note that this is the same image as used in the link further above.

**So there appears to be some sort of access issue, or url modification problem going on in Teams iOS when it tries to load the url as provided in the AdaptiveCard json payload, that only affects things stored in an Azure Blob Storage account...**

**Please help! Our AdaptiveCard on Teams iOS is useless at the moment as a result of this problem!**

_Reference - AdaptiveCard json payload_

`{
  "attachments": [
    {
      "content": {
        "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
        "body": [
          {
            "items": [
              {
                "size": "Medium",
                "text": "SOME TEXT HERE",
                "type": "TextBlock",
                "weight": "Bolder",
                "wrap": true
              },
              {
                "isSubtle": true,
                "size": "Small",
                "spacing": "Small",
                "text": "2021-04-16 13:40",
                "type": "TextBlock",
                "wrap": true
              }
            ],
            "type": "Container"
          },
          {
            "items": [
              {
                "horizontalAlignment": "Center",
                "msTeams": {
                  "allowExpand": true
                },
                "separator": true,
                "type": "Image",
                "url": "https://4rpm16vw6eyxvm4z05tvv4yr.blob.core.windows.net/99fpycl6ri99x2f/1000.png"
              }
            ],
            "type": "Container"
          },
          {
            "items": [
              {
                "size": "Medium",
                "spacing": "Medium",
                "text": "Text here:",
                "type": "TextBlock",
                "wrap": true
              },
              {
                "size": "Small",
                "spacing": "Small",
                "text": "SOME TEXT HERE",
                "type": "TextBlock",
                "wrap": true
              },
              {
                "size": "Small",
                "spacing": "Small",
                "text": "SOME TEXT HERE",
                "type": "TextBlock",
                "wrap": true
              }
            ],
            "type": "Container"
          }
        ],
        "type": "AdaptiveCard",
        "version": "1.2"
      },
      "contentType": "application/vnd.microsoft.card.adaptive",
      "contentUrl": null
    }
  ],
  "type": "message"
}`

**Comment by Wajeed-msft** [comment]
@Trinetra-MSFT , @Nikitha-MSFT - could you please look into this?

**Comment by aimskl** [comment]
Hi, 

We'd like to provide an update to the above in light of some further testing we've done on our side.

It turns out that the Azure Blob log entries mentioned above are not driven by the Teams iOS client, but instead from other clients (Teams Desktop and web browser). We previously did not have detailed Azure Monitor logging enabled so were not able to dig deeper into the related UserAgent strings. Now having gone through them, we can confirm that the entries do not correspond to the Teams iOS client. Though those blob read requests are triggering 304/ConditionNotMet errors, they are not preventing those clients from correctly loading the images in AdaptiveCards in Teams. _So we think these can be ignored for the purpose of this issue._

We learned two more interesting things:

- Attempting to view the AdaptiveCard image on Teams for iOS is not triggering any success/failure logs in Azure Monitor for Blob Storage. We tried this multiple times by isolating access to the blob images from only Teams iOS and no other clients. No logs entries were written.
- We took a look at the Teams iOS client logs (Settings -> Help and feedback -> Report an issue -> {the email attachments}). We found error logging there from the client's "PhotoMgr" module related to attempts to load the AdaptiveCard images. Snippets of the logs are pasted below. {we have also submitted these logs to the Teams iOS issue reporting email, with a reference to this issue posted on Github}

2021-04-25T02:53:47.603Z [I-AH]PhotoMgr offline or timeout for 1520392768 TSPhotoManager:1092
2021-04-25T02:53:47.603Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344
2021-04-25T02:53:47.619Z [I-AH][HTTP] GetUserGroupSettings : 200 TSHttpManager:351
2021-04-25T02:53:47.625Z [I-AH]PhotoMgr offline or timeout for 470724211 TSPhotoManager:1092
2021-04-25T02:53:47.625Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344
2021-04-25T02:53:47.625Z [I-AH]PhotoMgr offline or timeout for 470724211 TSPhotoManager:1092
2021-04-25T02:53:47.625Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344
2021-04-25T02:53:47.625Z [I-AH]PhotoMgr offline or timeout for 470724211 TSPhotoManager:1092
2021-04-25T02:53:47.625Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344
2021-04-25T02:53:47.625Z [I-AH]PhotoMgr offline or timeout for 1091472326 TSPhotoManager:1092
2021-04-25T02:53:47.625Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344
2021-04-25T02:53:47.653Z [I-AH]PhotoMgr offline or timeout for 1520392768 TSPhotoManager:1092
2021-04-25T02:53:47.653Z [W-AH][HTTP] PhotoMgr:CustomImgRenderer : 403, contextID:, domain:NSURLErrorDomain, code:403, MT ServerRequestID: TSHttpManager:344

2021-04-25T02:54:01.431Z [W-AH]PhotoMgr too many retries for 470724211 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:02.881Z [W-AH]PhotoMgr too many retries for 1520392768 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:02.881Z [W-AH]PhotoMgr too many retries for 1403742647 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:02.881Z [W-AH]PhotoMgr too many retries for 470724211 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:02.882Z [W-AH]PhotoMgr too many retries for 3738597851 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:03.247Z [W-AH]PhotoMgr too many retries for 898842238 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:03.248Z [W-AH]PhotoMgr too many retries for 1520392768 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:04.181Z [W-AH]PhotoMgr too many retries for 1091472326 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:04.181Z [W-AH]PhotoMgr too many retries for 470724211 (CustomImgRenderer) TSPhotoManager:923
2021-04-25T02:54:09.547Z [W-AH]PhotoMgr too many retries for 1091472326 (CustomImgRenderer) TSPhotoManager:923


Just to reiterate: the images in the AdaptiveCards load just fine when being viewed using Teams Desktop (PC or Mac) or Teams via browser. Since 2021-04-12 11:40 UTC, however, they don't load using the Teams iOS client. Prior to this timestamp, they loaded fine in Teams iOS. We have not changed anything from our side/config.

Is there somewhere else we can post this issue so that the Teams iOS client Team will be made aware of this problem/regression? 

Please help! Thank you

**Comment by aimskl** [comment]
@Trinetra-MSFT @Nikitha-MSFT @Wajeed-msft 

Please would someone help us on this. 

We are now just over two weeks with an effectively broken Teams AdaptiveCard image rendering client. I tried going directly to 365 customer support, but after they consulted with their "senior" people, they pointed me to Azure Support, whose automated problem detector refused to detect any issue, and which then told me I need to cough up $$$ for a support subscription. Can you imagine that ... here we are, long-time / multi-year M365 Business Premium and Azure Pay-As-You-Go subscription clients, Microsoft's Teams client code regresses and breaks our process, and we are expected to pay Microsoft to have the problem fixed. We have also submitted this problem directly through the Teams mobile app email channel, but have had zero response (not even an automated recognition / ticket system).

I just created a new test blob storage account and ran some simple cURL tests against a testing webhook in Teams. The problem is still there. Amazing.


SIMPLE STEPS TO REPRODUCE:

1. Create blob storage account. Create a container within. Set the access for that container to Anonymous Blob Access Only. 
2. Download the sample placeholder image found here: https://via.placeholder.com/1000 or use any you wish. Upload that image to your just created blob container.
3. Formulate the URL to that blob. Test it in your browser. Should work no problem (anonymously).
4. Create a test channel in a team in Teams. Create an incoming webhook to that channel and copy the webhook url
5. Assuming you are on Linux/Mac, open Terminal and use the below cURL command to send two separate cards. Send one using the url to the image you put in your blob container. Send another using the original url from where you got the placeholder image (e.g. https://via.placeholder.com/1000). (Be sure to escape the forward slashes as \/ )

curl -H 'Content-Type: application/json' -d '{"attachments":[{"content":{"$schema":"http:\/\/adaptivecards.io\/schemas\/adaptive-card.json","body":[{"items":[{"horizontalAlignment":"Center","msTeams":{"allowExpand":true},"separator":true,"type":"Image","url":"URL TO IMAGE GOES HERE"}],"type":"Container"}],"type":"AdaptiveCard","version":"1.2"},"contentType":"application\/vnd.microsoft.card.adaptive","contentUrl":null}],"type":"message"}' WEBHOOK URL GOES HERE (NOT QUOTED OR ESCAPED)

6. You will notice that the adaptivecard using the image from the source location you where got the placeholder works on any Teams client (iOS, Desktop, Web). But the card using the image stored in a public blob container only works on Teams Desktop and Web, and NOT on iOS.

As mentioned many times, this is Suddenly a new issue since April 12th. Prior to this date, this was working smoothly using the iOS client for nearly three years (!).

PLEASE PLEASE FIX! 

Thank you

**Comment by Trinetra-MSFT** [comment]
@aimskl , Thanks for raising this issue, Let me confirm this at my end, it seems it is a Bug, will raise it

**Comment by aimskl** [comment]
Hi Folks @Trinetra-MSFT @Nikitha-MSFT @Wajeed-msft 

This is still broken...and thus also our team. Amazing to think we are now considering returning to Slack.

**Comment by Wajeed-msft** [comment]
@aimskl - Thank you so much for sharing the Adaptive card JSON and the image. We are able to repro the issue at our end. 

![image](https://user-images.githubusercontent.com/31851992/117271680-0d74d080-ae78-11eb-871e-5e1878309018.png)

Internal Tracker: [Bug 1311239](https://domoreexp.visualstudio.com/MSTeams/_workitems/edit/1311239)

Could you please keep the test storage account till I share update on this?

**Comment by aimskl** [comment]
@Wajeed-msft Thank you, thank you so much for reproducing and actioning the bug for resolution. Sure, will leave the test storage account as is until you tell me otherwise.

**Comment by Wajeed-msft** [comment]
@aimskl - engineering team is looking into the bug, meanwhile just wanted to share possible workaround to unblock this scenario.

Given that issue is happening for image stored in Azure blob, alternative option to display image would be as Base64 image string. 

**Example JSON:** 

```
{
	"attachments": [
		{
			"content": {
				"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
				"body": [
					{
						"items": [
							{
								"size": "Medium",
								"text": "Embeded Image Test",
								"type": "TextBlock",
								"weight": "Bolder",
								"wrap": true
							},
							{
								"isSubtle": true,
								"size": "Small",
								"spacing": "Small",
								"text": "2021-04-16 13:40",
								"type": "TextBlock",
								"wrap": true
							}
						],
						"type": "Container"
					},
					{
						"items": [
							{
								"horizontalAlignment": "Center",
								"msTeams": {
									"allowExpand": true
								},
								"separator": true,
								"type": "Image",
								"url": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAA+gAAAPoBAMAAAC/jcnXAAAAG1BMVEXMzMyWlpacnJyqqqrFxcWxsbGjo6O3t7e+vr6He3KoAAAACXBIWXMAAA7EAAAOxAGVKw4bAAAKzklEQVR4nO3dSXPbRhoGYJBaj4Kc7ShmmfHRyjKVIxW7MjlaPqTmaGVS5Rwlp8pnKZnM/O1hg1s38JEitI4zz1NlUfooNTt4A6AbBIGqAgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA4Pau6rpV+eOn+i/ftn8tKu7/ODr85+WNX3m/rg82aDAshn1kU6ft0N/WyRfXF/dGqXb48qavvNMOPWwwLIZ9ZGNnrdC/q6eeXlfcHk1rh5c3fOXdVuhhg2Ex7CObq8vQ9+q582uKF/Paxzd85UEr9LDBqBh2h83ttUJfLOR8KUfF5ZK/6aK/KEMPGwyLYR/Z3G4Z+v5yIdeXa4tXy9onN3vpszL0sMGoGHaHHq7K0L/LFujn64rzfW1yOL7JK6fsstDDBsNi2Ed6OCtDP05L9x8v/kgL+8m64lZa5F/864dX6fHZTV55tww9bDAshn1kczt1EXqz5fx6Xr9cU0xbiCfjqpny3Wz7flqGHjYYFcPu0MNpGXpa+z5cPPFsTTFtIZq5884N17cmsiz0sMGoGHaHzf27LkM/WSzGtGH9YHVxexlC2tqOe7/y/lkZethgWAz7yIa2//Nl3Qp9smgPZ9+O5utTWEzL+2haG87Xxh5++HU6QluGHjYYFsM+spnlIHhZq5dT34vFso2Kg3oxcd5bJrOh5Zh8GXrYYFgM+8hmTrqh5/kN5ws8LOaLe9T3IMl23Q09bDAqht1hQ0HoW9mWeme+6wyLx9mG9bTvSC4KPWwwKobdYUNB6MNsTLY/X6PC4igbQl313chGoYcNRsWwO2woCP0kj2++xKPidr608xgazUx6nP3wUfnKQehhg2Ex7CMbCkI/zUfDx7OdaFTcybaxzQa33LOm6djr2fdpWt0KJgg9bDAshn1kQ8+/So7z0M/yZXgx24lGxd08573OnjX9//TJ8i/a+93t5pW/ykMPGwyLYR/pZZCHXmwt59vRqFhs0YutcGMr26SP2hv/hTz0sMGwGPaRXorQ6/bMeVWx+KvyN5Jm+33ZfJtW0BUHUPI/CxsMi2Ef6SVfbNvZrni6cR2vKJ4Ug7Ozzl77dLHyD1cPsPP4wgajYtgd+slD3y92v1vTtTUsXhX70uPO+2wp6ume97QzylvIQw8bjIphd+gnD70YLTc/na8oFkPo9FNrDJ026s3Odrs7YVvIQw8bjIphd+gnD72ce+1NF29YPG6fQtfebZ/N/mqrXv1uex562GBUDLtDP3nou50F+mxF8awdR3vidDXblS/fB+3KQw8bjIphd+inHfrl4pn96ZApLJ4V6+9VdxO+VS9W1pUjrVbo3QajYtgd+mmHPl48s70MvVMsx+sn3dCbSdu4CWXlO96t0LsNRsWwO/SThz6MFmhYHF0X+uxUpt3VE7Yy9LDBqBh2h37Wh360otiJo3NcLP3ZB82ufeXoem3ohyuKYXfo575CT0OsJ2kDvWrCJvTHk4feOqg5DSUs1sWB1/BgaDrm/vc1E7aykbDBqBh2h37uLfS0Zf9p7YxK6I/l3kJPY7h6zYRN6I/n3kKfnSmx5hRloT+W+xrITY/LrB1mGcg9lvsL/aQJfc3bIUJ/LPcXevNhtXXntQj9sdzTYdhktHbC5jDs47mnN1yS42tGWa3QveHyYNa/n/5sRfH4urdWq9nwfd3nC/PQwwajYtgd+mmfObNcoLNzVMLiaRHHaRju1jXT9PaZM90Go2LYHfrJQy/PQ5kt3rB4cc3pUslVE/qa9TAPPWwwKobdoZ889Ls7MTI5a0Jf86mjPPSwwajoxMg7kId+d6dAV4srwK1+k80p0I+mOKxZDLfnz0TF8mBoOEofTkNfsyLmfxY2GBbDPtJLsdiKoyEnN/9YUzLZBdcXa2dUeXw+1vSQitCPow8HRsVi4hTOm5prBKWt7+pPleahhw2GxbCP9FKEfhF9DDgqFmPo7keVZ8UPUvKr18Q89LDBsBj2kV6K0E/u5qIEs7+arJjH6+bReeguSvCQitCH2cCruPxIpzjKZmlX0ep83GR0sm7SVgzJwgajYtgdeilCv7sLDaU4nlxzOcki9LDBqOhCQ7dXhH53lxSbXXRk9pmHWBF62GBUdEmx2+vMhbML89Wri4Pl4t6LpukXs/XxdM2krTPj7jQYFsM+0kcZ+ml0Cc6ouLNc38LLhI5ma3h6cpNPrcYNhsWwj/RRhn5SBxfbjYrXXBA4pdXEkVbQTT6f7oLAD6kMff5h0+Ky2mFxMRkLx2oni1V0tHq3W+4WwgajYtgd+ihDby77l5Zy5yL/7eLVfNGfRhvwRVjNbvcofuky9LDBqBh2hz5ab1mktG5/O4/sSNzykvwdZehu5/FwWqHfzY17smPuxSVDS2XobtzzcFqh380tuq6ybfpZsCWYKkN3i66H035H+mKxPG9xM76zrLj6SGwrdDfjezDt0O/itpuLC4ol+SVDS63Q3XbzwXTOPflttjyfXlfcn+1vuzfYHeYZZZcMbWmHHjYYFsM+cgvNSLl9m+qouNPkcfNbaXeEDYbFsI/cwu/RDemj4v7fgtvZ30bYYFgM+wgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAP87BrN//B8R+p9Wffj1imdaoR9MfhrUdV0dVtXwIP048Xz0pqq+r3+uBkdVNbrfnnJnDra/WvFMEHrz9aPL6t0s9O0vxr9Pvrx4NR78Uu2d3X9vuRMH1XfVl59Vw/He5JudT59Vg1cH6WHr09PJs4NXn1W755NnstA/fl29mYW+9XL6ZefZ4F21e/GY/x30MAl95+fvX26dn1S/Vj+++KYa/Nw8/PgirbiDyVP7R9U3VRb6wdP9o1now/H0y/bRYHj5mwHA+2KyeR+Ot4/2n706/7b6fLJOD8bNw+fVyeTZweSp6s3k32Tnn/bnTejvds9noQ/mXw4Ge6/fCP19MRnINSvw0dPXRynYlGF6OJjv0w+q37bOq3xNH76tumt69dcjob8vDlJqaXV+/cvryerdJJsesjV9+Gr2i7PQdz6suvv06vSl0N8Xk/DSPr16df72ZfVu/Dwlmx6W+/Rq7+PZL85Cb35ovmaj98qk/v2RwpuM3qur8cm42hs9TdGlh+Xovdo7mv1iHvp0F/C8ns/TK6H/qeyeP3YPeHBvH7sDPLjB08fuAQAAAHfkv/elJK03/YyFAAAAAElFTkSuQmCC"
							}
						],
						"type": "Container"
					},
					{
						"items": [
							{
								"size": "Medium",
								"spacing": "Medium",
								"text": "Text here:",
								"type": "TextBlock",
								"wrap": true
							},
							{
								"size": "Small",
								"spacing": "Small",
								"text": "SOME TEXT HERE",
								"type": "TextBlock",
								"wrap": true
							},
							{
								"size": "Small",
								"spacing": "Small",
								"text": "SOME TEXT HERE",
								"type": "TextBlock",
								"wrap": true
							}
						],
						"type": "Container"
					}
				],
				"type": "AdaptiveCard",
				"version": "1.2"
			},
			"contentType": "application/vnd.microsoft.card.adaptive",
			"contentUrl": null
		}
	],
	"type": "message"
}
```

**Output:**
![image](https://user-images.githubusercontent.com/31851992/117399050-45821f00-af1d-11eb-810b-c7e9be38ed1e.png)

We will continue with fixing the main issue. Let me know if this unblocks for now.

**Comment by aimskl** [comment]
Hi @Wajeed-msft 

Appreciate the efforts to help, but the Base64 approach for the image won't work for us. We explored this avenue a few years ago but were restricted by payload size limits.

Though our image size varies depending on the time of day, it is generally around 228 KB, or ~240k characters in Base64. 

Also, our process uses Logic Apps and the Teams connector to post to the Teams channel. This is the error received:
`{
  "error": "The payload is too large. Please make sure the size is less than 28KB."
}`
This is a known limitation documented here: https://docs.microsoft.com/en-gb/connectors/teams/#known-issues-and-limitations

The example image you used is ~3.4 KB and only ~4k characters in Base64.

**Comment by Wajeed-msft** [comment]
@aimskl - Updates on the investigation:  Engineering team has identified the root cause of the issue and working on finding an appropriate solution.

**Comment by Wajeed-msft** [comment]
@aimskl - Engineering team has identified the fix and we verified that it's working fine on iOS device. Currently fix is undergoing review process.

**Comment by aimskl** [comment]
Thank you for the update @Wajeed-msft

**Comment by Wajeed-msft** [comment]
@aimskl - we will keep this open till the deployment rolls out to public store build.

**Comment by Nikitha-MSFT** [comment]
Bug is Resolved now, we have tested this at our end,Hence, we are closing this question for now. If you still face any issue please feel free to reopen the issue

**Comment by aimskl** [comment]
@Nikitha-MSFT are you serious? The matter is not resolved on our end. We don't see any fix, nor any update in the public app store. Why would you close a ticket before verifying with the client whether or not they have received the fix and that it works on their side? 

Yet another example of rather tone-deaf Microsoft and customer communications. Just to get you guys to pay some attention to me here, I had to write emails to your senior executives. Does anyone at Microsoft seriously consider an efficient way to for their client's to spend their time?

Why would you pre-empt your customer? I will provide feedback on this for sure.

**Comment by aimskl** [comment]
@Wajeed-msft @Nikitha-MSFT @Trinetra-MSFT 

I can confirm that our team received the updated Teams app from the iOS store today and the issue mentioned above is now resolved on our end. We will go ahead and disable the testing storage accounts used to demonstrate the above.

Thank you to all who helped to get this fixed and turned around. Much appreciated.

**Comment by Wajeed-msft** [comment]
@aimskl - apologies for closing this without confirming in the first place. Glad to know it's working for you now. Please let me know if you are facing any other issues.
